src/citrine/resources/ingestion.py

```diff
@@ -200,6 +217,70 @@ def status(self) -> IngestionStatus:
         return IngestionStatus.build(self.session.get_resource(path=path))
 
 
+class FailedIngestion(Ingestion):
```

Comment:
I'm not immediately understanding the FailedIngestion class and why it inherits from Ingestion. 😅  Everything looks good and logical, just hoping to get an explanation!

Reply:
The shapes of the responses under different types of errors are different -- the errors returned when a file upload files are structured differently than when there's a missing ingredient, for example.  

The `FailedIngestion` is essentially written to unify the interfaces so that no matter how your ingest failed, you end up with a consistent object to query.  When you get a failure early in the process, this class makes assumptions how to enrich that to look like the later failures.